### PR TITLE
YSQL: Add `attributes (jsonb)` to `yb_tablet_metadata`

### PR DESCRIPTION
**Pull request by keisku**
Closes https://github.com/yugabyte/yugabyte-db/issues/29665

---

**Testing**

Upgrade ysql.

```bash
./bin/yb-ctl --rf 3 create
./build/latest/bin/yb-admin upgrade_ysql
```

Able to see `active_ssts_size` and `wals_size` in the `yb_tablet_metadata`.

```bash
./bin/ysqlsh -c "\d yb_tablet_metadata"
            View "pg_catalog.yb_tablet_metadata"
     Column      |  Type   | Collation | Nullable | Default
-----------------+---------+-----------+----------+---------
 tablet_id       | text    |           |          |
 oid             | oid     |           |          |
 db_name         | text    |           |          |
 relname         | text    |           |          |
 start_hash_code | integer |           |          |
 end_hash_code   | integer |           |          |
 leader          | text    |           |          |
 replicas        | text[]  |           |          |
 attributes      | jsonb   |           |          |
```

Run this right after creating a cluster

```bash
./bin/ysqlsh -c "
SELECT *
FROM yb_tablet_metadata
WHERE relname = 'test_sizes';
"
 tablet_id | oid | db_name | relname | start_hash_code | end_hash_code | leader | replicas | attributes
-----------+-----+---------+---------+-----------------+---------------+--------+----------+------------
(0 rows)
```

Create a test table

```bash
./bin/ysqlsh -c "CREATE TABLE test_sizes (k INT PRIMARY KEY, v TEXT);"
CREATE TABLE
```

See values in `attributes`.

```bash
./bin/ysqlsh -P pager=off -c "
SELECT *
FROM yb_tablet_metadata
WHERE relname = 'test_sizes';
"
            tablet_id             |  oid  | db_name  |  relname   | start_hash_code | end_hash_code |     leader     |                    replicas                    |                                                                                                    attributes
----------------------------------+-------+----------+------------+-----------------+---------------+----------------+------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 5fdb14d4b4f547729f3d18153910137b | 16384 | yugabyte | test_sizes |               0 |         65536 | 127.0.0.1:5433 | {127.0.0.1:5433,127.0.0.2:5433,127.0.0.3:5433} | {"replicas": {"127.0.0.1:5433": {"wal_sizes": 1048576, "active_sst_sizes": 0}, "127.0.0.2:5433": {"wal_sizes": 1048576, "active_sst_sizes": 0}, "127.0.0.3:5433": {"wal_sizes": 1048576, "active_sst_sizes": 0}}}
(1 row)
```

Checked the UI.

13-40-v0q3y-86duc

Insert 29000 rows.

```bash
./bin/ysqlsh -c "
INSERT INTO test_sizes
SELECT g, repeat('x', 100000)
FROM generate_series(1001, 30000) g;
"
INSERT 0 29000
```

`active_ssts_size` and `wals_size` in the `attributes` increased.

```bash
./bin/ysqlsh -P pager=off -c "
SELECT tablet_id, attributes
FROM yb_tablet_metadata
WHERE relname = 'test_sizes';
"
            tablet_id             |                                                                                                            attributes
----------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 140d3712996b4b3182d9fc1f897a3fbc | {"replicas": {"127.0.0.1:5433": {"wal_sizes": 0, "active_sst_sizes": 0}, "127.0.0.2:5433": {"wal_sizes": 2902590966, "active_sst_sizes": 139142511}, "127.0.0.3:5433": {"wal_sizes": 2902590911, "active_sst_sizes": 139142511}}}
 da0d353355d6449da1a10714fcbc6c54 | {"replicas": {"127.0.0.1:5433": {"wal_sizes": 0, "active_sst_sizes": 0}, "127.0.0.2:5433": {"wal_sizes": 2902590966, "active_sst_sizes": 139142511}, "127.0.0.3:5433": {"wal_sizes": 2902590911, "active_sst_sizes": 139142511}}}
(2 rows)
```

Checked the UI.

13-42-zp1ls-31lse

---

> [!NOTE]
> **Medium Risk**
> Changes the `yb_get_tablet_metadata` system function/view signature and adds a new migration, which could impact clients or tests that depend on the previous column set. Also extends master-to-pggate protobuf plumbing for per-replica size metrics, so mismatched versions or edge cases around missing metrics could surface.
> 
> **Overview**
> Adds an `attributes` (`jsonb`) column to `yb_tablet_metadata` (and underlying `yb_get_tablet_metadata()` record) that returns per-replica `active_sst_sizes` and `wal_sizes` keyed by replica address.
> 
> Plumbs replica size data from master tablet metadata through protobuf (`TabletStatusPB`), pggate descriptors, and Postgres JSONB construction, and includes a new YSQL migration plus updated regression tests and docs/examples to reflect the new column.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 79d4847ed6f0098f4560cb0ddd4c5069a6c4b723. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/yb/yql/pgwrapper/pg_mini-test.cc [hunk 1]
```diff
@@ -3304,6 +3304,56 @@ TEST_F(PgMiniTest, TabletMetadataOidMatchesPgClass) {
...
+          "       array_length(active_ssts_size, 1),"
+          "       array_length(wals_size, 1) "
+          "FROM yb_tablet_metadata WHERE relname = 'size_test'")));
+  ASSERT_EQ(replicas_len, ssts_len);
+  ASSERT_EQ(replicas_len, wals_len);
+  ASSERT_EQ(static_cast<size_t>(replicas_len), NumTabletServers());
+
+  // Unnest all three arrays in parallel to verify ordering and alignment.
+  // Uses WITH ORDINALITY to join corresponding elements in a single snapshot.
+  auto result = ASSERT_RESULT(pg_conn.Fetch(
```
Comment (jasonyb) [inline src/yb/yql/pgwrapper/pg_mini-test.cc]: use `auto rows = ...FetchRows<...>(`

File: src/yb/yql/pgwrapper/pg_mini-test.cc [hunk 2]
```diff
@@ -3304,6 +3304,56 @@ TEST_F(PgMiniTest, TabletMetadataOidMatchesPgClass) {
...
+      "FROM yb_tablet_metadata tm, "
+      "     unnest(tm.replicas)          WITH ORDINALITY AS r(val, ord), "
+      "     unnest(tm.active_ssts_size)  WITH ORDINALITY AS s(val, ord), "
+      "     unnest(tm.wals_size)         WITH ORDINALITY AS w(val, ord) "
+      "WHERE tm.relname = 'size_test' "
+      "  AND r.ord = s.ord AND r.ord = w.ord "
+      "ORDER BY r.ord"));
+
+  auto num_rows = PQntuples(result.get());
+  ASSERT_EQ(static_cast<size_t>(num_rows), NumTabletServers());
```
Comment (jasonyb) [inline src/yb/yql/pgwrapper/pg_mini-test.cc]: Use `rows.size()`

File: src/yb/yql/pgwrapper/pg_mini-test.cc [hunk 3]
```diff
@@ -3304,6 +3304,56 @@ TEST_F(PgMiniTest, TabletMetadataOidMatchesPgClass) {
...
+
+  // Validate per-replica rows to catch index misalignment that array-length checks cannot detect.
+  for (int i = 0; i < num_rows; ++i) {
```
Comment (jasonyb) [inline src/yb/yql/pgwrapper/pg_mini-test.cc]: `for (const auto [replica, sst_size, wal_size] : rows) {`

File: src/yb/yql/pgwrapper/pg_mini-test.cc [hunk 4]
```diff
@@ -3304,6 +3304,56 @@ TEST_F(PgMiniTest, TabletMetadataOidMatchesPgClass) {
...
+    auto replica  = ASSERT_RESULT(GetValue<std::string>(result.get(), i, 0));
+    auto sst_size = ASSERT_RESULT(GetValue<int64_t>(result.get(), i, 1));
+    auto wal_size = ASSERT_RESULT(GetValue<int64_t>(result.get(), i, 2));
+    ASSERT_FALSE(replica.empty());
+    ASSERT_GE(sst_size, 0) << "replica " << replica;
+    ASSERT_GE(wal_size, 0) << "replica " << replica;
+  }
+
+  // Every replica address must correspond to a live tserver.
+  auto valid_count = ASSERT_RESULT(pg_conn.FetchRow<int64_t>(
```
Comment (jasonyb) [inline src/yb/yql/pgwrapper/pg_mini-test.cc]:
```suggestion
  auto valid_count = ASSERT_RESULT(pg_conn.FetchRow<PGUint64>(
```

File: src/postgres/src/include/catalog/pg_proc.dat [hunk 5]
```diff
@@ -12234,9 +12234,9 @@
   proname => 'yb_get_tablet_metadata', prorows => '100',
   proretset => 't', provolatile => 'v', proparallel => 'r',
   prorettype => 'record', proargtypes => '',
-  proallargtypes => '{text,text,text,text,text,int4,int4,text,_text}',
-  proargnames => '{tablet_id,object_uuid,namespace,object_name,type,start_hash_code,end_hash_code,leader,replicas}',
-  proargmodes => '{o,o,o,o,o,o,o,o,o}',
+  proallargtypes => '{text,text,text,text,text,int4,int4,text,_text,_int8,_int8}',
+  proargnames => '{tablet_id,object_uuid,namespace,object_name,type,start_hash_code,end_hash_code,leader,replicas,active_ssts_size,wals_size}',
```
Comment (jasonyb) [inline src/postgres/src/include/catalog/pg_proc.dat]: these are arrays of foo size, so they should be named foo_sizes.  `active_sst_sizes`, `wal_sizes`.  `wals_size` sounds like the sum total size of all wals.

File: src/yb/yql/pgwrapper/ysql_migrations/V97__29665__yb_tablet_metadata_sizes.sql [hunk 6]
```diff
@@ -0,0 +1,42 @@
+BEGIN;
+  SET LOCAL yb_non_ddl_txn_for_sys_tables_allowed TO true;
+
+  -- Add active_ssts_size and wals_size columns to yb_get_tablet_metadata().
+  UPDATE pg_catalog.pg_proc SET
+    proallargtypes = '{25,25,25,25,25,23,23,25,1009,1016,1016}',
+    proargmodes = '{o,o,o,o,o,o,o,o,o,o,o}',
+    proargnames = '{tablet_id,object_uuid,namespace,object_name,type,start_hash_code,end_hash_code,leader,replicas,active_ssts_size,wals_size}'
+  WHERE oid = 8099;
```
Comment (jasonyb) [inline src/yb/yql/pgwrapper/ysql_migrations/V97__29665__yb_tablet_metadata_sizes.sql]: Although less efficient, I feel it is better to use `WHERE proname = 'yb_get_tablet_metadata' AND pronamespace = 11;` which is resilient to potential oid changes

File: src/yb/yql/pggate/ybc_pggate.cc [hunk 7]
```diff
@@ -2992,21 +2992,38 @@ YbcStatus YBCTabletsMetadata(YbcPgGlobalTabletsDescriptor** tablets, size_t* cou
...
 
       new (dest) YbcPgGlobalTabletsDescriptor {
         .tablet_descriptor = MakeYbcPgTabletsDescriptor(tablet_metadata),
         .replicas = replicas_array,
-        .replicas_count = static_cast<size_t>(tablet_metadata.replicas().size()),
-        .is_hash_partitioned = tablet_metadata.is_hash_partitioned()
+        .replicas_count = static_cast<size_t>(replicas_size),
+        .is_hash_partitioned = tablet_metadata.is_hash_partitioned(),
+        .replicas_sst_files_size = sst_sizes_array,
+        .replicas_wal_files_size = wal_sizes_array
```
Comment (jasonyb) [inline src/yb/yql/pggate/ybc_pggate.cc]:
```suggestion
        .replica_sst_sizes = sst_sizes_array,
        .replica_wal_sizes = wal_sizes_array
```

File: src/yb/yql/pggate/ybc_pggate.cc [hunk 8]
```diff
@@ -2992,21 +2992,38 @@ YbcStatus YBCTabletsMetadata(YbcPgGlobalTabletsDescriptor** tablets, size_t* cou
 
     for (const auto& tablet_metadata : tablet_metadatas) {
       const char** replicas_array = nullptr;
+      uint64_t* sst_sizes_array = nullptr;
+      uint64_t* wal_sizes_array = nullptr;
+      const auto replicas_size = tablet_metadata.replicas().size();
```
Comment (jasonyb) [inline src/yb/yql/pggate/ybc_pggate.cc]:
```suggestion
      const auto replicas_count = tablet_metadata.replicas().size();
```

File: src/yb/yql/pggate/ybc_pg_typedefs.h [hunk 9]
```diff
@@ -854,6 +854,8 @@ typedef struct {
   const char** replicas;
   size_t replicas_count;
   bool is_hash_partitioned;
+  uint64_t* replicas_sst_files_size;
+  uint64_t* replicas_wal_files_size;
```
Comment (jasonyb) [inline src/yb/yql/pggate/ybc_pg_typedefs.h]:
```suggestion
  const uint64_t* replica_sst_sizes;
  const uint64_t* replica_wal_sizes;
```

File: src/yb/yql/pgwrapper/pg_mini-test.cc [hunk 10]
```diff
@@ -3304,6 +3304,56 @@ TEST_F(PgMiniTest, TabletMetadataOidMatchesPgClass) {
       << " but yb_tablet_metadata returned " << tablet_metadata_oid;
 }
 
+TEST_F(PgMiniTest, TabletMetadataSizeColumns) {
```
Comment (jasonyb) [inline src/yb/yql/pgwrapper/pg_mini-test.cc]: I believe this test can preferably be rewritten as a regress test in, for example, `src/postgres/src/test/regress/expected/yb.orig.tablet_metadata.out`.  Any queries that select nondeterministic data can be written differently, such as `ASSERT_GE(sst_size, 0) << "replica " << replica;` transformed to selecting `s.val > 0 AS sst_size`, or if the number of replicas is nondeterministic, put the opposite condition in the where clause and expect zero rows returned.

File: src/yb/tablet/tablet.proto [hunk 11]
```diff
@@ -76,6 +76,9 @@ message TabletStatusPB {
   // Leader is the last replica in the list.
   repeated string replicas = 24;
   optional bool is_hash_partitioned = 25;
+  // Per-replica SST and WAL sizes in bytes, parallel to the replicas array.
+  repeated uint64 replicas_sst_files_size = 26;
+  repeated uint64 replicas_wal_files_size = 27;
```
Comment (jasonyb) [inline src/yb/tablet/tablet.proto]: replica_sst_sizes, replica_wal_sizes

File: src/postgres/src/backend/utils/misc/pg_yb_utils.c [hunk 12]
```diff
@@ -8496,31 +8504,69 @@ yb_get_tablet_metadata(PG_FUNCTION_ARGS)
...
+			{
+				int			si = sort_idx[idx];
+
+				text_elems[idx] = CStringGetTextDatum(tablet->replicas[si]);
+				sst_elems[idx] = Int64GetDatum(
+					tablet->replicas_sst_files_size
+						? (int64) tablet->replicas_sst_files_size[si] : 0);
+				wal_elems[idx] = Int64GetDatum(
+					tablet->replicas_wal_files_size
+						? (int64) tablet->replicas_wal_files_size[si] : 0);
```
Comment (jasonyb) [inline src/postgres/src/backend/utils/misc/pg_yb_utils.c]: UInt64GetDatum

File: src/postgres/src/backend/utils/misc/pg_yb_utils.c [hunk 13]
```diff
@@ -8496,31 +8504,69 @@ yb_get_tablet_metadata(PG_FUNCTION_ARGS)
...
 
 			/*
-			 * Sort the list lexicographically for consistency, so that all rows
-			 * with same replicas have same entries.
+			 * Build a sort index so replicas and their sizes stay parallel
+			 * after lexicographic sorting.
 			 */
-			list_sort(replicas_list, string_list_compare);
-			values[8] = PointerGetDatum(strlist_to_textarray(replicas_list));
+			sort_idx = palloc(nreplicas * sizeof(int));
```
Comment (jasonyb) [inline src/postgres/src/backend/utils/misc/pg_yb_utils.c]:
```suggestion
			sort_idxs = palloc(nreplicas * sizeof(int));
```

**Review by jasonyb — COMMENTED** [review]
docs/content/stable/explore/observability/yb-tablet-metadata.md:

```
diff --git a/docs/content/stable/explore/observability/yb-tablet-metadata.md b/docs/content/stable/explore/observability/yb-tablet-metadata.md
index 112465c01b..9963e1a468 100644
--- a/docs/content/stable/explore/observability/yb-tablet-metadata.md
+++ b/docs/content/stable/explore/observability/yb-tablet-metadata.md
@@ -37,6 +37,8 @@ The following table describes the columns of the `yb_tablet_metadata` view.
 | end_hash_code | int | Ending hash code (exclusive) for the tablet. (NULL for range-sharded tables.) |
 | leader | text | IP address, port of the leader node for the tablet. |
 | replicas | text[] | A list of replica IP addresses and port (includes leader) associated with the tablet. |
+| active_ssts_size | bigint[] | Per-replica SST files size in bytes, in the same order as replicas. |
+| wals_size | bigint[] | Per-replica WAL files size in bytes, in the same order as replicas. |
 
 ## Examples
 
```

yb.port.rules.out

```
diff --git a/src/postgres/src/test/regress/expected/yb.orig.tablet_metadata.out b/src/postgres/src/test/regress/expected/yb.orig.tablet_metadata.out
index b6981e2f4f..29f230dc80 100644
--- a/src/postgres/src/test/regress/expected/yb.orig.tablet_metadata.out
+++ b/src/postgres/src/test/regress/expected/yb.orig.tablet_metadata.out
@@ -15,6 +15,21 @@ ORDER BY start_hash_code NULLS FIRST;
  test_table_1 | yugabyte |           32768 |         65536
 (3 rows)
 
+-- Test that active_ssts_size and wals_size columns are present and are arrays
+SELECT
+    relname,
+    pg_typeof(active_ssts_size) AS sst_type,
+    pg_typeof(wals_size) AS wal_type,
+    array_length(active_ssts_size, 1) = array_length(replicas, 1) AS sizes_match_replicas
+FROM yb_tablet_metadata
+WHERE relname = 'test_table_1'
+ORDER BY start_hash_code NULLS FIRST
+LIMIT 1;
+   relname    | sst_type | wal_type | sizes_match_replicas 
+--------------+----------+----------+----------------------
+ test_table_1 | bigint[] | bigint[] | t
+(1 row)
+
 -- Test that we are able to join with yb_servers()
 SELECT
     ytm.relname,
```

**Comment by hbhanawat** [comment]
@keisku Instead of adding two fields, we would want a single json column for all the tablet attributes. This will ensure that the other tablet properties could be added in future without modifying the schema of the table cc: @IshanChhangani @GauravSingh-yb @jasonyb

**Comment by keisku** [comment]
@hbhanawat @jasonyb 

I've addressed review points and updated the PR title/description accordingly 👍

**Comment by keisku** [comment]
@jasonyb any chance you could take a look at this PR again?

File: src/postgres/src/backend/utils/misc/pg_yb_utils.c [hunk 14]
```diff
@@ -8512,31 +8520,103 @@ yb_get_tablet_metadata(PG_FUNCTION_ARGS)
...
+			 * Build a sort index so replicas stay in lexicographic order.
 			 */
-			list_sort(replicas_list, string_list_compare);
-			values[8] = PointerGetDatum(strlist_to_textarray(replicas_list));
+			sort_idxs = palloc(nreplicas * sizeof(int));
+			for (size_t idx = 0; idx < nreplicas; idx++)
+				sort_idxs[idx] = (int) idx;
+
+			qsort_arg(sort_idxs, nreplicas, sizeof(int),
+					  replica_index_cmp, tablet->replicas);
```
Comment (jasonyb) [inline src/postgres/src/backend/utils/misc/pg_yb_utils.c]: given jsonb does not preserve ordering, you can leave the `replicas` code untouched and do the `attributes` code separately

File: src/yb/yql/pgwrapper/ysql_migrations/V101__29665__yb_tablet_metadata_sizes.sql [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+    FROM
+        yb_get_tablet_metadata() t
+    LEFT JOIN
+        pg_class c ON c.relname = t.object_name
+    LEFT JOIN
+        pg_namespace n ON n.oid = c.relnamespace
+    WHERE
+        (t.namespace = 'system' AND t.object_name = 'transactions')
+    OR
+        (t.type = 'YSQL' AND n.nspname NOT IN ('pg_catalog', 'information_schema'));
```
[the remark this answers is not in the document]
Comment (jasonyb) [reply src/yb/yql/pgwrapper/ysql_migrations/V101__29665__yb_tablet_metadata_sizes.sql:41]: it is further outdated by this point.  needs V100 at least.  Update to latest master.

File: src/postgres/src/test/regress/expected/yb.orig.tablet_metadata.out [hunk 16]
```diff
@@ -15,6 +15,54 @@ ORDER BY start_hash_code NULLS FIRST;
...
+ replica 
+---------
+(0 rows)
+
+-- Test that every replica address corresponds to a live tserver
+SELECT unnest(replicas) AS orphan_replica
+FROM yb_tablet_metadata
+WHERE relname = 'test_table_1'
+EXCEPT
+SELECT host || ':' || port FROM yb_servers();
```
Comment (jasonyb) [inline src/postgres/src/test/regress/expected/yb.orig.tablet_metadata.out]: this does not necessarily mean all replicas are present. it could be an empty result or just one host from yb_tablet_metadata

File: src/postgres/src/backend/utils/misc/pg_yb_utils.c [hunk 17]
```diff
@@ -8548,31 +8552,104 @@ yb_get_tablet_metadata(PG_FUNCTION_ARGS)
...
 			Assert(tablet->replicas != NULL);
 
 			/* The last replica is the leader. */
-			values[7] = CStringGetTextDatum(tablet->replicas[tablet->replicas_count - 1]);
+			values[7] = CStringGetTextDatum(tablet->replicas[nreplicas - 1]);
+
+			/*
+			 * Build JSONB attributes before sorting replicas, because the
+			 * parallel size arrays must stay aligned with the replica order.
+			 * JSONB does not preserve key order so sorting is unnecessary.
```
Comment (jasonyb) [inline src/postgres/src/backend/utils/misc/pg_yb_utils.c]: What is "parallel size arrays"?  I don't see a reason why you need to handle `attributes` before `replicas`.  The sort is on a local list variable.  If there is no strong reason, the order should be the expected values[8] then [9]

Comment (keisku) [reply src/postgres/src/backend/utils/misc/pg_yb_utils.c]: Sorry for the confusion. I missed this in my self-review. Fixed order to values[8], values[9] and removed the misleading comment.

File: src/postgres/src/backend/utils/misc/pg_yb_utils.c [hunk 18]
```diff
@@ -8564,31 +8568,100 @@ yb_get_tablet_metadata(PG_FUNCTION_ARGS)
...
+
+					/* Per-replica object */
+					pushJsonbValue(&jb_state, WJB_BEGIN_OBJECT, NULL);
+
+					jb_key.type = jbvString;
+					jb_key.val.string.val = "active_sst_sizes";
+					jb_key.val.string.len = 16;
+					pushJsonbValue(&jb_state, WJB_KEY, &jb_key);
+					jb_val.type = jbvNumeric;
+					jb_val.val.numeric = DatumGetNumeric(
```
Comment (jasonyb) [inline src/postgres/src/backend/utils/misc/pg_yb_utils.c:8638]: Error  (S&RX) bad_opening_paren
    There should be no linebreak after (

File: src/postgres/src/backend/utils/misc/pg_yb_utils.c [hunk 19]
```diff
@@ -8564,31 +8568,100 @@ yb_get_tablet_metadata(PG_FUNCTION_ARGS)
...
+						DirectFunctionCall1(int8_numeric, Int64GetDatum(sst_size)));
+					pushJsonbValue(&jb_state, WJB_VALUE, &jb_val);
+
+					jb_key.type = jbvString;
+					jb_key.val.string.val = "wal_sizes";
+					jb_key.val.string.len = 9;
+					pushJsonbValue(&jb_state, WJB_KEY, &jb_key);
+					jb_val.type = jbvNumeric;
+					jb_val.val.numeric = DatumGetNumeric(
```
Comment (jasonyb) [inline src/postgres/src/backend/utils/misc/pg_yb_utils.c:8647]: Error  (S&RX) bad_opening_paren
    There should be no linebreak after (

File: src/postgres/src/backend/utils/misc/pg_yb_utils.c [hunk 20]
```diff
@@ -118,6 +118,7 @@
 #endif
 #include "storage/procarray.h"
 #include "tcop/utility.h"
+#include "utils/array.h"
```
Comment (jasonyb) [inline src/postgres/src/backend/utils/misc/pg_yb_utils.c:121]:
```suggestion
```
